ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/services/admin/ApiPackageManagementService.scala

```diff
@@ -74,30 +77,45 @@ class ApiPackageManagementService(
         FutureConverters
           .toScala(res._1)
           .flatMap {
-            case UploadPackagesResult.Ok =>
+            //TODO BH: first step to get this to compile -- need to implement properly polling and retrieval of response event
+            case SubmissionResult.Acknowledged =>
               Future.successful(UploadDarFileResponse())
-            case r @ UploadPackagesResult.Overloaded =>
-              Future.failed(ErrorFactories.resourceExhausted(r.description))
-            case r @ UploadPackagesResult.InternalError(_) =>
+            case r @ SubmissionResult.InternalError(_) =>
               Future.failed(ErrorFactories.internal(r.reason))
-            case r @ UploadPackagesResult.InvalidPackage(_) =>
-              Future.failed(ErrorFactories.invalidArgument(r.description))
-            case r @ UploadPackagesResult.ParticipantNotAuthorized =>
-              Future.failed(ErrorFactories.permissionDenied(r.description))
-            case r @ UploadPackagesResult.NotSupported =>
-              Future.failed(ErrorFactories.unimplemented(r.description))
+            case SubmissionResult.NotSupported =>
+              Future.failed(ErrorFactories.unimplemented("not supported"))
+            case SubmissionResult.Overloaded =>
+              Future.failed(ErrorFactories.grpcError(io.grpc.Status.RESOURCE_EXHAUSTED))
```

Comment:
Please use
`Future.failed(ErrorFactories.resourceExhausted(r.description))` instead

Reply:
Changed to use this -- had to add description to SubmissionResult case objects as they did not have this field previously